cr8/engine.py

```diff
@@ -73,6 +72,15 @@ def __init__(self, hosts, concurrency, sample_mode, session_settings=None):
         self.concurrency = concurrency
         self.client = client(hosts, session_settings=session_settings, concurrency=concurrency)
         self.sampler = get_sampler(sample_mode)
+        self.session_settings = session_settings
+
+    def init(self):
+        if isinstance(self.client, HttpClient):
+            f = self.client.execute
+            for setting, value in self.session_settings.items():
+                stmt = f'set {setting}={value}'
+                statements = itertools.repeat((stmt, ()), self.concurrency)
+                aio.run_many(f, statements, self.concurrency, self.concurrency)
```

Comment:
@mfussenegger Do you have any opinion on this approach? I think this needs some cleanup, but would you be ok with this solution?

Reply:
Besides breaking encapsulation I think this might become unrealiable with a higher concurrency. The first set statements can finish before laters are triggered - which means a connection could get re-used instead of creating a new one for each `set`

I think I'd rather have a minimal extra pool implementation in the client on top of the session, and setting limit=1 on the underlying connector to ensure there is no double pooling.

Reply:
ok, makes sense. Thanks for the feedback.

Reply:
> I think I'd rather have a minimal extra pool implementation in the client on top of the session, and setting limit=1 on the underlying connector to ensure there is no double pooling.

Do you mean a pool with multiple instances of `TCPConnector`?

Reply:
This would mean I would need to create also multiple `ClientSessions`, one for each `TCPConnector` and pool them too.

Reply:
A custom `Connector` where I maintain the connections is not an option, because I need to have a ClientSession to run HTTP operations on top of this.

Reply:
I can try the option of using multiple `ClientSessions` each having it's own `TCPConnector` with limit 1 and see how the performance is.